test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smtrcomp.py

```diff
@@ -439,6 +447,55 @@ def test_inference_compiler_neo(self, sagemaker_session, ecr_image, framework_ve
         estimator.compile_model(target_instance_family='ml_p3',
                                 input_shape={'data':[1, 28, 28]},
                                 output_path=s3_prefix,
-                                framework='keras',
-                                framework_version='2.6.0',
+                                framework='tensorflow',
+                                framework_version='2.9',
                                 )
+
+
+@pytest.mark.integration("trcomp")
+class TestUserExperience:
+
+    @pytest.fixture()
+    def instance_type(self):
+        return 'ml.p3.2xlarge'
+
+
+    @pytest.fixture()
+    def instance_count(self):
+        return 1
+
+
+    @pytest.mark.xfail(reason="Signature based failure attribution does not differentiate between XLA errors and TF/Py errors. Stack trace based failure attribution does not work with TF since the stack trace is manipulate to abstract away graph mode execution details.")
```

Comment:
Out of curiosity, when is this test expected to fail?

Reply:
Got it. Would it be better to skip this test until the training toolkit propagates the error appropriately?

Reply:
`skip` or `xfail` are semantically equivalent (apart from resource usage). I'd prefer not changing the code if not improving things.

Reply:
`xfail` isn't represented properly on CodeBuild test reports, from my past experience. While this is okay if an engineer is reviewing the test job results manually, this lack of support for `xfail` reports will result in blockers in using automation to collect/review the test results.

It would also be nice to not have to spend time trying, waiting, and failing on these tests if we know they will fail, because the tests are all run sequentially, rather than in parallel.